### PR TITLE
[M1026] Support transferring Benthic Photo Quadrat Transect (Image Classification) back to Collect Record

### PR DESCRIPTION
**Pull request by gridcell**
## Summary by CodeRabbit

- **New Features**
	- Added a new Boolean field `image_classification` to enhance data tracking for the `benthicphotoquadrattransect` model.
	- Introduced methods to handle image classification data in the `BenthicPhotoQuadratTransectProtocolWriter` class.

- **Bug Fixes**
	- Streamlined the submission process by removing version distinctions in the `CollectRecordViewSet`.

- **Tests**
	- Improved test coverage for the `CollectRecord` functionality.
	- Added assertions to ensure project revision numbers are updated correctly after operations.

- **Chores**
	- Enhanced code organization and readability in various mixins and methods.

### CODE REVIEW

**Comment by gridcell** [comment]
@coderabbitai review

**Comment by gridcell** [comment]
[ML PQT conversion back to Collect Records](https://trello.com/c/dqAArLZE/1026-ml-pqt-conversion-back-to-collect-records)

File: src/api/resources/sampleunitmethods/benthicphotoquadrattransectmethod.py [hunk 1]
```diff
@@ -81,6 +81,26 @@ class BenthicPhotoQuadratTransectMethodView(
     serializer_class = BenthicPhotoQuadratTransectMethodSerializer
     http_method_names = ["get", "put", "head", "delete"]
 
+    def edit_sample_unit(self, request, pk):
+        bpqt_record = self.get_queryset().get(pk=pk)
```
Comment (saanobhaai) [inline src/api/resources/sampleunitmethods/benthicphotoquadrattransectmethod.py:85]: Wondering if we should protect against a NotFound exception here or further upstream

File: src/api/resources/sampleunitmethods/benthicphotoquadrattransectmethod.py [hunk 2]
```diff
@@ -81,6 +81,26 @@ class BenthicPhotoQuadratTransectMethodView(
...
+        image_classification = bpqt_record.image_classification
+        cr = super().edit_sample_unit(request, pk)
+        if image_classification is True:
+            image_classification = True
```
Comment (saanobhaai) [inline src/api/resources/sampleunitmethods/benthicphotoquadrattransectmethod.py:89]: What CodeRabbit said

File: src/api/submission/writer.py [hunk 3]
```diff
@@ -407,18 +409,79 @@ def get_or_create_benthic_photo_quadrat_transect(
...
+                    "benthic_photo_quadrat_transect": benthic_photo_quadrat_transect_id,
+                    "image_id": image_id,
+                    "attribute": attribute_id,
+                    "growth_form": growth_form_id,
+                    "quadrat_number": quadrat_num,
+                    "num_points": count,
+                }
+            )
+            if image_id not in images:
+                quadrat_num += 1
```
Comment (saanobhaai) [inline src/api/submission/writer.py:469]: This logic looks great, but I'm just pausing before merging to consider one more time whether there's any chance that this quadrat num assignment will be confusing/inconsistent for users submitting and unsubmitting.

File: src/api/tests/test_sample_unit_method.py [hunk 4]
```diff
@@ -70,29 +70,6 @@ def test_edit_belt_fish_transect_method(
...
-    assert CollectRecord.objects.filter(id=collect_record_id).exists() is False
-    assert AuditRecord.objects.filter(
-        record_id=collect_record_id,
-        event_type=AuditRecord.SUBMIT_RECORD_EVENT_TYPE,
-        model=CollectRecord.__name__.lower(),
-    ).exists()
-    assert BeltFish.objects.filter(id=collect_record4.data.get("sample_unit_method_id")).exists()
-
-
 def test_submit_collect_record_v2(
```
Comment (saanobhaai) [inline src/api/tests/test_sample_unit_method.py:73]: rename?

**Review by saanobhaai — CHANGES_REQUESTED** [review]
Looks great, just a couple of things to pause on before merging

File: src/api/resources/sampleunitmethods/benthicphotoquadrattransectmethod.py (hunk 2 above)
Comment (gridcell) [reply src/api/resources/sampleunitmethods/benthicphotoquadrattransectmethod.py:89]: I want to force `image_classification` True or False and remove None if it exists.

I will change the line `if image_classification is True:` to `if image_classification`

**Review by saanobhaai — APPROVED** [review]

**Pull request merged**